### PR TITLE
add header for PREP() macro

### LINKED ISSUE #1021: CBA PREP() macro could be useful thing to have documented
**CBA Version:**  stable

**Description:**
For example ACE mentions this about the PREP macro:

> Functions shall be created in the functions\ subdirectory, named fnc_functionName.sqf They shall then be indexed via the PREP(functionName) macro in the XEH_preInit.sqf file.
> 
> The PREP macro allows for CBA function caching, which drastically speeds up load times. Beware though that function caching is enabled by default and as such to disable it you need to #define DISABLE_COMPILE_CACHE above your #include "script_components.hpp" include!

As it is useful outside CBA, maybe it should be documented also on the wiki and or here:
http://cbateam.github.io/CBA_A3/docs/files/main/script_macros_common-hpp.html

### PR DESCRIPTION
**Pull request by commy2**
**When merged this pull request will:**
- title
- close #1021

### CODE REVIEW

File: addons/main/script_macros_common.hpp [hunk 1]
```diff
@@ -890,6 +890,39 @@ Macro: GVARMAIN()
...
+    Full file path:
+        '\MAINPREFIX\PREFIX\SUBPREFIX\COMPONENT\fnc_<FNC>.sqf'
+
+    Resulting function name:
+        'PREFIX_COMPONENT_<FNC>'
+
+    The PREP macro should be placed in a script run by a XEH preStart and XEH preInit event.
+
+    The PREP macro allows for CBA function caching, which drastically speeds up load times.
+    Beware though that function caching is enabled by default and as such to disable it you, need to
```
Comment (jonpas) [inline addons/main/script_macros_common.hpp]: to disable it, you need

Comment (commy2) [reply addons/main/script_macros_common.hpp]: fixed

File: addons/main/script_macros_common.hpp [hunk 2]
```diff
@@ -936,6 +962,13 @@ Macro: PATHTO_FNC()
 Author:
     dixon13, commy2
  ------------------------------------------- */
+#ifdef RECOMPILE
```
Comment (dedmen) [inline addons/main/script_macros_common.hpp]: Now the PATHTO_FNC header looks like it applies to recompile.

Comment (commy2) [reply addons/main/script_macros_common.hpp]: The alternative is having `RECOMPILE` to look like it belongs to `PREP` when it belongs to `PATHTO_FNC` only.

Comment (dedmen) [reply addons/main/script_macros_common.hpp]: maybe move it below PATHTO_FNC then? So that the header is still directly above what it belongs to

Comment (commy2) [reply addons/main/script_macros_common.hpp]: `PATHTO_FNC` uses `RECOMPILE`. I think it would break the way you described.

Comment (dedmen) [reply addons/main/script_macros_common.hpp]: Macro inside a different macro is only resolved when the second macro is used.
PATHTO_FNC is resolved first. Then the RECOMPILE inside it is resolved.
So it wouldn't cause any problems if it's defined after the PATHTO_FNC macro

**Pull request merged**